### PR TITLE
Fix rust agent header and empty query string

### PR DESCRIPTION
**Pull request by taiyosogawa**
Cleans up the Rust http_client

### Changes proposed: 
- Ensure that the rust SDK version is included in the user agent header
- Fix a rust bug that appended `?` to URLs, even when there is no query string

### CODE REVIEW

File: rs/src/management/http_client.rs [hunk 1]
```diff
@@ -518,6 +499,33 @@ impl From<TunnelClientBuilder> for TunnelManagementClient {
     }
 }
 
+fn add_query(mut url: Url, tunnel_opts: &TunnelRequestOptions) -> Url
+{
```
Comment (connor4312) [inline rs/src/management/http_client.rs:503]: Please make sure to rust `rustfmt` for formatting. Turning on "format on save" with the rust analyzer extension will do this for you automatically, or you can manually run the "Format Document" command

File: rs/src/management/http_client.rs [hunk 2]
```diff
@@ -472,12 +451,14 @@ pub struct TunnelClientBuilder {
 /// Creates a new tunnel client builder. You can set options, then use `into()`
 /// to get the client instance (or cast automatically).
 pub fn new_tunnel_management(user_agent: &str) -> TunnelClientBuilder {
+    let pkg_version = PKG_VERSION.unwrap_or("unknown");
+    let full_user_agent = format!("{}{}{}", user_agent, " Visual-Studio-Tunnel-Service-Rust-SDK/", pkg_version);
```
Comment (connor4312) [inline rs/src/management/http_client.rs:455]: Probably want to put a space or some kind of separator after the consumer-provided user agent.

File: rs/src/management/http_client.rs [hunk 3]
```diff
@@ -518,6 +499,33 @@ impl From<TunnelClientBuilder> for TunnelManagementClient {
...
+        url.query_pairs_mut().append_pair("forceRename", "true");
+    }
+    if !tunnel_opts.tags.is_empty() {
+        url.query_pairs_mut().append_pair("tags", &tunnel_opts.tags.join(","));
+        if tunnel_opts.require_all_tags {
+            url.query_pairs_mut().append_pair("allTags", "true");
+        }
+    }
+    if tunnel_opts.limit > 0 {
+        url.query_pairs_mut().append_pair("limit", &tunnel_opts.limit.to_string());
```
Comment (connor4312) [inline rs/src/management/http_client.rs:523]: If this function took a `&mut Url`, you could keep the query pairs reference around as the original code did without having to return the Url. Not a big deal just less verbosity.

File: rs/src/management/http_client.rs (hunk 1 above)
Comment (taiyosogawa) [reply rs/src/management/http_client.rs]: Good call! I ran the formatter and added a note to CONTRIBUTING.md.

File: rs/src/management/http_client.rs (hunk 3 above)
Comment (taiyosogawa) [reply rs/src/management/http_client.rs]: Thanks for the tip - this is my first rust change and the feedback is appreciated. Done!

**Review by connor4312 — APPROVED** [review]

**Pull request merged**